component/remote/s3/s3.go

```diff
@@ -135,7 +135,9 @@ func generateS3Config(args Arguments) (*aws.Config, error) {
 	// Override the endpoint.
 	if args.Options.Endpoint != "" {
 		endFunc := aws.EndpointResolverWithOptionsFunc(func(service, region string, _ ...interface{}) (aws.Endpoint, error) {
-			return aws.Endpoint{URL: args.Options.Endpoint}, nil
+			// The S3 compatible system used for testing with does not require signing region, so it's fine to be blank
+			// but when using a proxy to real S3 it needs to be injected.
+			return aws.Endpoint{URL: args.Options.Endpoint, SigningRegion: args.Options.SigningRegion}, nil
```

Comment:
In what situations will signing_region != region?

Reply:
Where you are authing against one region but accessing a bucket in another. Of which I can only find vague references to. AWS itself does this internally for their endpoint.
```
// aws s3 code

if len(endpoint.SigningRegion) != 0 {
   ctx = awsmiddleware.SetSigningRegion(ctx, endpoint.SigningRegion)
} else {
   ctx = awsmiddleware.SetSigningRegion(ctx, resolveRegion)
}
```

I could embed the above pivoting but I think I want to keep it specific and unambiguous when using a custom endpoint.